src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -283,6 +283,71 @@ void Snippet::initSupportedPrimitiveDescriptors() {
 }
 
 void Snippet::selectOptimalPrimitiveDescriptor() {
+    // Check if memory desc is in SupportedPrimitiveDescriptors.
+    auto memDescInSupportedPD = [&](const ov::intel_cpu::MemoryDescPtr& parentMemDesc, const size_t& idx) {
+        for (auto& type : getImplPriority()) {
+            for (size_t i = 0; i < getSupportedPrimitiveDescriptors().size(); i++) {
+                const auto& supportedPrimitiveDesc = getSupportedPrimitiveDescriptors()[i];
+                const impl_desc_type supportedType = supportedPrimitiveDesc.getImplementationType();
+                if (supportedType == type) {
+                    const auto& supportedMemDesc = supportedPrimitiveDesc.getConfig().inConfs[idx].getMemDesc();
+                    if (parentMemDesc->getShape().isStatic() &&
+                        parentMemDesc->getShape() == supportedMemDesc->getShape() &&
+                        parentMemDesc->getPrecision() == supportedMemDesc->getPrecision()) {
+                        //
+                        if ((parentMemDesc->hasLayoutType(LayoutType::ncsp) &&
+                             supportedMemDesc->hasLayoutType(LayoutType::ncsp)) ||
+                            (parentMemDesc->hasLayoutType(LayoutType::nspc) &&
+                             supportedMemDesc->hasLayoutType(LayoutType::nspc))) {
+                            return true;
+                        }
+                    }
+                }
+            }
+        }
+        return false;
+    };
+
+    if (getParentEdges().size() == 3u) {
+        auto check_special_shape = [](const ov::intel_cpu::Shape& shape) {
+            // SShape=[1,?,1,1]
+            if (shape.isStatic()) {
+                const auto& dims = shape.getDims();
+                if (dims.size() == 4u) {
+                    return dims[0] == 1u && dims[1] > 1u && dims[2] == 1u && dims[3] == 1u;
+                }
+            }
+            return false;
+        };
+
+        bool foundOptimalPD = true;
+        for (size_t i = 0; i < getParentEdges().size(); i++) {
+            auto parentEdge = getParentEdgeAt(i);
+            auto parentPtr = parentEdge->getParent();
+            auto parent_spd = parentPtr->getSelectedPrimitiveDescriptor();
+            auto parentDesc = parent_spd->getConfig().outConfs[0].getMemDesc();
+            if (i == 0u || i == 2u) {
```

Comment:
Minor: We can significantly improve readability by using `switch(i)` instead of the `if() + break` clauses here.
The only drawback of the `switch` is that we can't `break` the loop from the inside, but we can modify the `for` condition to `i < getParentEdges().size() && foundOptimalPD` for example.
Even if we don't `break` the loop, it's not a big deal, since there are only 3 iterations.

Reply:
@IvanNovoselov , I found I can't limit parameters specific position, I just statistic num_sshape_ncsp == 2 && num_nonsshape_nspc == 1, if it is met, I think optimal PD is found.